server/tests/src/test/java/org/infinispan/server/cli/CliIT.java

```diff
@@ -94,25 +101,46 @@ public void testCliInteractive() {
    }
 
    @Test
-   public void testCliBatch() {
-      AeshTestShell shell = new AeshTestShell();
-      CLI.main(shell, new String[]{"-f", this.getClass().getResource("/cli/batch.cli").getPath()});
-      shell.assertContains("Hi CLI");
-      shell.assertContains("batch1");
+   public void testCliBatch() throws IOException {
+      // Create batch file programmatically so that we can supply the correct server address for the connect command:
+      Path testPath = Paths.get(CommonsTestingUtil.tmpDirectory(CliIT.class));
+      testPath.toFile().mkdirs();
+      try {
+         Path batchFilePath = testPath.resolve("batch.cli").toAbsolutePath();
+         try (PrintWriter out = new PrintWriter(Files.newBufferedWriter(batchFilePath))) {
+            out.println("echo Hi CLI");
+            out.printf("connect %s:11222\n", hostAddress());
```

Comment:
Interesting. Makes me think we should support env and property replacement in batches

Reply:
That would be great :slightly_smiling_face:

Reply:
https://github.com/infinispan/infinispan/pull/8435

Reply:
Rebased on your commit. Much simpler